src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,685 @@
+        sendMissingTrustedTimeSource(commandPath.mEndpointId);
+    }
+
+    TimeSynchronizationServer::Instance().SetTrustedTimeSource(tts);
+    commandObj->AddStatus(commandPath, status);
+    return true;
+}
+
+bool emberAfTimeSynchronizationClusterSetTimeZoneCallback(
+    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::TimeSynchronization::Commands::SetTimeZone::DecodableType & commandData)
+{
+    auto timeZone = commandData.timeZone;
+    size_t items;
+    uint8_t maxSize = 0;
+    timeZone.ComputeSize(&items);
+    TimeZoneListMaxSize::Get(commandPath.mEndpointId, &maxSize);
+
+    if (items > maxSize || items > CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE)
+    {
+        commandObj->AddStatus(commandPath, Status::ResourceExhausted);
+        return true;
+    }
+    // first element shall have validAt entry of 0
+    // if second element, it shall have validAt entry of non-0
+
+    if (CHIP_NO_ERROR != TimeSynchronizationServer::Instance().SetTimeZone(timeZone))
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+    sendTimeZoneStatus(commandPath.mEndpointId, 0);
+    sendTimeFailure(commandPath.mEndpointId); // TODO remove
+    GetDelegate()->HandleTimeZoneChanged(TimeSynchronizationServer::Instance().GetTimeZone());
+    GetDelegate()->HandleDstoffsetlookup();
+
+    TimeSynchronization::TimeZoneDatabaseEnum tzDb;
+    TimeZoneDatabase::Get(commandPath.mEndpointId, &tzDb);
+    if (GetDelegate()->HasFeature(0, TimeSynchronization::TimeSynchronizationFeature::kTimeZone) &&
+        tzDb != TimeSynchronization::TimeZoneDatabaseEnum::kNone)
+    {
+        auto tz = TimeSynchronizationServer::Instance().GetTimeZone().begin();
+        Commands::SetTimeZoneResponse::Type response;
+        if (GetDelegate()->HandleDstoffsetavailable(tz->name.Value()) == true)
+        {
+            GetDelegate()->HandleGetdstoffset();
+            response.DSTOffsetRequired = false;
+            sendDSTStatus(commandPath.mEndpointId, true);
+        }
+        else
+        {
+            TimeSynchronizationServer::Instance().ClearDSTOffset();
+            sendDSTTableEmpty(commandPath.mEndpointId);
+            response.DSTOffsetRequired = true;
+            sendDSTStatus(commandPath.mEndpointId, false);
+        }
+        commandObj->AddResponse(commandPath, response);
+        computeLocalTime(commandPath.mEndpointId);
+        return true;
+    }
+
+    computeLocalTime(commandPath.mEndpointId);
+
+    commandObj->AddStatus(commandPath, Status::Success);
+
+    return true;
+}
+
+bool emberAfTimeSynchronizationClusterSetDSTOffsetCallback(
+    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::TimeSynchronization::Commands::SetDSTOffset::DecodableType & commandData)
+{
+    Status status  = Status::Success;
+    auto dstOffset = commandData.DSTOffset;
```

Comment:
```suggestion
    const auto & dstOffset = commandData.DSTOffset;
```
and please just do this everywhere.